crates/bvs-vault-router/src/contract.rs

```diff
@@ -325,13 +326,135 @@ mod execute {
                     .add_attribute("reason", data.metadata.reason),
             ))
     }
+
+    pub fn slash_locking(
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        id: state::SlashingRequestId,
+    ) -> Result<Response, ContractError> {
+        let registry = state::get_registry(deps.storage)?;
+
+        let slash_req = state::SLASHING_REQUESTS.may_load(deps.storage, id.clone())?;
+
+        if slash_req.is_none() {
+            return Err(ContractError::InvalidSlashingRequest {
+                msg: "Slashing entry with given id does not exist".to_string(),
+            });
+        }
+
+        let slash_req = slash_req.unwrap();
```

Comment:
nit but can make it more rusty by using match
```suggestion
        let slash_req = match slash_req {
            Some(req) => req,
            None => {
                return Err(ContractError::InvalidSlashingRequest {
                    msg: "Slashing entry with given id does not exist".to_string(),
                });
            }
        };
```